### DISCUSSION TITLE
Trimming Map

### DISCUSSION THREAD

**Discussion by rodolfomachado** [humbertogontijo/homeassistant-roborock#476] [General]
Guys,

I using [lovelace-xiaomi-vacuum-map-card](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card) and I want to zoom in the map. 

Looking ther I found that I need to trim the map in map extractor. 

https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/discussions/227

Since I am only using this integration how can I trim the map?

**Comment by bobzer** [comment]
if you can use the crop option of the card or the  camera.proxy 
I didn't get perfect result with both of them
also normally in integration you can click on configure and camera and the crop , but that didn't do anything for me

**Reply by rodolfomachado** [reply]
EDIT: I was trying to use other options apart from crop as someone said it uses a lot of cpu.

Could you explain about camera.proxy?

As you said in integration > configure > camera does nothing to me. I also tried to use "map scale" but did nothing.

ah, btw, after I played integration > configure > camera> seems that the integration stopped updating properly but I have to do more testing

**Reply by bobzer** [reply]
you're right after the configure thing nothing is update anymore,  you need to refresh the integration 

on you're screenshot, the problem is alignment, crop need to be at the same level than camera. (personnally any performance problem)

about camera proxy i don't know much, : https://www.home-assistant.io/integrations/proxy/
i tried it but not long enough and i don't understant the result really well because on the preview it seems to works but when open it doesn't...so if you have a big screen it doesn;t works maybe it's my config i don't know 
exemple of what i put in configuration.yml:

```
camera:
  - platform: proxy
    entity_id: camera.kirby_map
    name: kirby_map_crop
    mode: crop
    image_top: 120
    image_left: 110
    max_image_width: 315
    max_image_height: 140
```

**Reply by rodolfomachado** [reply]
You're right about crop. I realized that and edited the post but you've not seen I guess.

I adjust that and it is working. Not that well but acceptable.

As I can see camera proxy will do the same as crop so I will stick to crop at this moment.

Tks
